independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -253,44 +253,55 @@ public InputStream getResourceAsStream(String nm) {
 
     @Override
     protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
-        ClassLoaderState state = getState();
-        synchronized (getClassLoadingLock(name)) {
-            Class<?> c = findLoadedClass(name);
-            if (c != null) {
-                return c;
-            }
-            String resourceName = sanitizeName(name).replace(".", "/") + ".class";
-            boolean parentFirst = parentFirst(resourceName, state);
-            if (state.bannedResources.contains(resourceName)) {
-                throw new ClassNotFoundException(name);
-            }
-            if (parentFirst) {
-                try {
-                    return parent.loadClass(name);
-                } catch (ClassNotFoundException ignore) {
-                    log.tracef("Class %s not found in parent first load from %s", name, parent);
+        //even if the thread is interrupted we still want to be able to load classes
+        //if the interrupt bit is set then we clear it and restore it at the end
+        boolean interrupted = Thread.interrupted();
+        try {
```

Comment:
In general I think this makes sense.  It's not perfect in that the thread can be interrupted *during* the class load, but I don't really have any better solution for that at bootstrap (`JBossThread` does have support for blocking interrupts but I'm assuming that it isn't available at bootstrap).

Reply:
I guess we could catch InterruptedIOException and retry if it fails?

Reply:
Yeah that could work.  That exception has an `int` field which contains the number of bytes transferred thus far.  You'll have to clear the interrupt status manually though because `InterruptedIOException` (unlike `InterruptedException`) does not imply this.